public_html/wp-content/plugins/camptix-attendee-survey/camptix-attendee-survey.php

```diff
@@ -0,0 +1,159 @@
+<?php
+/**
+ * Plugin Name:     CampTix Attendee Survey
+ * Plugin URI:      https://wordcamp.org
+ * Description:     Send survey to WordCamp attendees.
+ * Author:          WordCamp.org
+ * Author URI:      https://wordcamp.org
+ * Version:         1
+ *
+ * @package         CampTix\AttendeeSurvey
+ */
+
+namespace CampTix\AttendeeSurvey;
+
+use function CampTix\AttendeeSurvey\Email\{add_email, delete_email};
+use function CampTix\AttendeeSurvey\Page\{add_page, delete_page};
+
+defined( 'WPINC' ) || die();
+
+/**
+ * Local dependencies.
+ */
+require_once get_includes_path() . 'email.php';
+require_once get_includes_path() . 'page.php';
+
+/**
+ * Plugin activation and deactivation hooks.
+ */
+register_deactivation_hook( __FILE__, __NAMESPACE__ . '\deactivate' );
+
+/**
+ * Actions & hooks
+ */
+add_action( 'plugins_loaded', __NAMESPACE__ . '\load' );
+
+/**
+ * Get the ID of the survey feature.
+ */
+function get_feature_id() {
+	return 'attendee_survey';
+}
+
+/**
+ * Check dependencies for loading the plugin.
+ *
+ * @return bool
+ */
+function can_load() {
+	$skip_feature = wcorg_skip_feature( get_feature_id() );
+
+	return ! $skip_feature;
+}
+
+/**
+ * Include the rest of the plugin.
+ *
+ * @return void
+ */
+function load() {
+	if ( ! can_load() ) {
+		return;
+	}
+
+	// We only want to admin panel on central, nothing else.
+	if ( WORDCAMP_ROOT_BLOG_ID === get_current_blog_id() ) {
+		require_once get_includes_path() . 'admin-page.php';
+	}
+
+	if ( is_wordcamp_type( 'next-gen' ) ) {
+		require_once get_includes_path() . 'cron.php';
+		add_action( 'init', __NAMESPACE__ . '\activate_on_current_site' );
+	}
+}
+
+/**
+ * The activation routine for a single site.
+ *
+ * @return void
+ */
+function activate_on_current_site() {
+	add_page();
+	add_email();
+
+	// Flushing the rewrite rules is buggy in the context of `switch_to_blog`.
+	// The rules will automatically get recreated on the next request to the site.
+	delete_option( 'rewrite_rules' );
+}
+
+/**
+ * Remove the survey page.
+ *
+ * @param bool $is_network True if deactivating network-wide.
+ *
+ * @return void
+ */
+function deactivate( $is_network = false ) {
+	if ( $is_network ) {
+		deactivate_on_network();
+	} else {
+		deactivate_on_current_site();
+	}
+}
+
+/**
+ * Run the deactivation routine on all valid sites in the network.
+ *
+ * @return void
+ */
+function deactivate_on_network() {
+	$valid_sites = get_site_ids_without_skip_flag();
+
+	foreach ( $valid_sites as $blog_id ) {
+		switch_to_blog( $blog_id );
+		deactivate_on_current_site();
+		restore_current_blog();
+	}
+}
+
+/**
+ * The deactivation routine for a single site.
+ *
+ * @return void
+ */
+function deactivate_on_current_site() {
+	delete_page();
+	delete_email();
+}
+
+/**
+ * Get the IDs of sites that do not have the FEATURE_ID skip feature flag.
+ *
+ * @return array
+ */
+function get_site_ids_without_skip_flag() {
```

Comment:
Since we've needed this here and in the Speaker Feedback plugin, I think it makes sense to extract it into a generic function in `mu-plugins/wp-cli-commands/miscellaneous.php`, next to `set_skip_feature_flag()`.

That way any plugin can use it, rather than duplicating it.

Reply:
I also wondered about just removing it since we're limiting it to events and this survey will probably apply across the board.

I think I'll remove it. Yep.